### PR TITLE
Add volume constraint check

### PR DESCRIPTION
**Pull request by arthurcgc**
This pull request makes tsuru api check the user's pool for the available volume-plans before creating a volume.
It also displays the avaiable volume-plans for the user when running: `tsuru pool constraint list`

### CODE REVIEW

File: api/volume.go [hunk 1]
```diff
@@ -126,6 +126,15 @@ func volumeCreate(w http.ResponseWriter, r *http.Request, t auth.Token) (err err
 	if !canCreate {
 		return permission.ErrUnauthorized
 	}
+
+	err = servicemanager.Volume.CheckPoolVolumeConstraints(ctx, inputVolume)
```
Comment (arthurcgc) [inline api/volume.go:130]: Create test for CheckVolumeConstraints

File: provision/pool/pool.go [hunk 2]
```diff
@@ -436,6 +459,10 @@ func ListPoolsForTeam(ctx context.Context, team string) ([]Pool, error) {
 	return getPoolsSatisfyConstraints(ctx, true, ConstraintTypeTeam, team)
 }
 
+func ListPoolsForVolumePlan(ctx context.Context, volumePlanName string) ([]Pool, error) {
```
Comment (arthurcgc) [inline provision/pool/pool.go:462]: Create test for ListPoolsForVolumePlan

File: provision/pool/pool_test.go [hunk 3]
```diff
@@ -654,15 +654,16 @@ func (s *S) TestPoolAllowedValues(c *check.C) {
 	constraints, err := pool.allowedValues()
```
Comment (arthurcgc) [inline provision/pool/pool_test.go:654]: consider adding some checks inside provision/pool/pool_test.go since volumePlanNames was added

**Review by cezarsa — APPROVED** [review]

File: types/volume/volume.go [hunk 4]
```diff
@@ -17,6 +17,7 @@ var (
 	ErrVolumeAlreadyBound       = errors.New("volume already bound in mountpoint")
 	ErrVolumeBindNotFound       = errors.New("volume bind not found")
 	ErrVolumeAlreadyProvisioned = errors.New("updating a volume already provisioned is not supported, a new volume must be created and the old one deleted if necessary")
+	ErrVolumePlanNotFound       = errors.New("volume-plan not present in pool constraint")
```
Comment (wpjunior) [inline types/volume/volume.go:20]: @arthurcgc `volume-plan is not present in pool constraint`

**Review by wpjunior — APPROVED** [review]

**Pull request merged**